cmd/s2i/main.go

```diff
@@ -228,7 +229,8 @@ func newCmdRebuild(cfg *api.Config) *cobra.Command {
 
 			// Attempt to read the .dockercfg and extract the authentication for
 			// docker pull
-			if r, err := os.Open(cfg.DockerCfgPath); err == nil {
+			r, err = os.Open(cfg.DockerCfgPath)
```

Comment:
Hmm, wait... did you notice that we're opening the same file twice? Is that even necessary?! 215 and 231 are opening the same file.. at first sight it doesn't look like the code in between the two calls would actually change the config file, would it?!